src/compiler/llvm/mir.rs

```diff
@@ -29,6 +29,32 @@ use super::llvmir::{get_ptr_alignment, LlvmIsAggregateType, LlvmToBasicTypeEnum}
 
 const ADDRESS_SPACE: AddressSpace = AddressSpace::Generic;
 
+#[derive(Clone, Copy)]
+pub enum Location<'ctx> {
```

Comment:
Maybe change this to address?  Not happy with the name.

Also, need to add docs.